hugo/content/docs/model_sources.md

```diff
@@ -0,0 +1,146 @@
+DiagramServerProxy <-.->|Action| DiagramServer
+{{< /mermaid>}}
+
+The following sections will explain how to use and work with them.
+
+## general usage
+regardless of the model source we are using, the first thing we have to do is always register it in the frontend DI-container like this:
+```Typescript
+bind(TYPES.ModelSource).to(ModelSourceClassOrProxy).inSingletonScope();
+```
+After that, we can retrieve it with the following code to further configure and use it.
+```Typescript
+const modelSource = container.get<LocalModelSource>(TYPES.ModelSource);
+```
+
+
+## Local Model Source
+The `LocalModelSource` allows to set and modify the model through function calls and keeps the model schema saved locally.
+To see how to use this model source, let's have a look at the following example:
+```Typescript
+import {SNode} from 'sprotty-protocol';
+export default runExample() {
+    const container = createContainer('sprotty-showcase');
+    const modelSource = container.get<LocalModelSource>(TYPES.ModelSource);
+    modelSource.setModel({
+        type: 'graph'
+        children: [
+            <SNode>{
+            type: 'node',
+                id: 'main_node',
+                text: 'node1',
+                position: {x: 0, y: 0}
+            }
+        ]
+    });
+
+    document.getElementById('addButton').onClick(() => {
+        modelSource.addElements([{parentId: 'graph', {
+            type: 'node',
+            id: 'new_node',
+            text: 'new node',
+            position: {x: 100, y: 100}
+            }}])
+    })
+}
+```
+In this example, we have a hard-coded data structure containing all our initial elements which are set as the model. 
+Be aware that by defining the model this way we are defining the model schema and not the actual model. Because of that, we should use the types for our nodes, edges, etc. from `sprotty-protocol` and not the `sprotty` main package.
+
+After defining the model schema we can then use methods like `addElements()` from the `LocalDataSource` to add new nodes at the click of a button. The `LocalDataSource` then handles updating the model and notifying the `ActionDispatcher` about the update so that the view can receive an animated update.
+
+Through methods like this, the `LocalModelSource` can also be used as a facade over the action-based API of Sprotty. It handles actions for bounds calculation and model updates.
+
+## Diagram Server
+When having the source based on which we want to generate the Diagram remotely, like in a worker or on a remote server, we can use Sprotty's DiagramServer. It communicates with the client through so-called `Action` objects which can be serialized to plain JSON.
+
+On client-side, instead of registering an actual `ModelSource` we use a `DiagramServerProxy`. The Proxy handles the communication and forwards actions to the `ActionDipatcher`. Out of the box, Sprotty offers the `WebSocketDiagramServerProxy` for communicating through WebSockets with the `DiagramServer`.
```

Comment:
```suggestion
On the client-side, instead of registering an actual `ModelSource` we can use a `DiagramServerProxy`. The Proxy handles the communication and forwards actions to the `ActionDipatcher`. Out of the box, Sprotty offers the `WebSocketDiagramServerProxy` for communicating through WebSockets with the `DiagramServer`.
```